flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/PushWatermarkIntoTableSourceScanRuleBase.java

```diff
@@ -117,8 +126,33 @@ protected FlinkLogicalTableSourceScan getNewScan(
                 idleTimeoutMillis = -1L;
             }
 
+            Optional<RelHint> optionsHintOptional =
+                    scan.getHints().stream()
+                            .filter(
+                                    relHint ->
+                                            relHint.hintName.equalsIgnoreCase(
+                                                    FlinkHints.HINT_NAME_OPTIONS))
+                            .findFirst();
+            Configuration hintOptions = new Configuration();
+            if (optionsHintOptional.isPresent()) {
+                RelHint optionsHint = optionsHintOptional.get();
+                hintOptions = Configuration.fromMap(optionsHint.kvOptions);
+            }
```

Comment:
I think we can easily avoid to create unnecessary objects via 
~~~java
Configuration hintOptions =
        optionsHintOptional
                .map(relHint -> Configuration.fromMap(relHint.kvOptions))
                .orElseGet(Configuration::new);
~~~

Reply:
modified